projects/addon-charts/components/line-chart/line-chart.component.ts

```diff
@@ -160,10 +162,11 @@ export class TuiLineChartComponent {
         return `${this.autoIdString}_${index}`;
     }
 
-    protected getImplicit($implicit: TuiPoint): TuiPoint | readonly TuiPoint[] {
+    protected getImplicit($implicit: TuiPoint): readonly TuiPoint[] {
         return (
-            this.hintDirective?.getContext(this.value.indexOf($implicit), this) ||
-            $implicit
+            this.hintDirective?.getContext(this.value.indexOf($implicit), this) || [
+                $implicit,
```

Comment:
This way if we fallback to `$implicit` we will recreate the array each time. Let's use `tuiMapper` in template?